src/detail-page/components/TabJobContent/FetchJobContent.tsx

```diff
@@ -0,0 +1,79 @@
+import { Button, CircularProgress, Input } from '@material-ui/core';
+import styled from 'styled-components';
+import * as React from 'react';
+import { Redirect } from 'react-router-dom';
+import { Job } from '../../../data/ITProfessions';
+import FetchJobElement from './FetchJobElement';
+import hiringPicture from '../../../icons/hiring.svg';
+
+const SearchInput = styled(Input)`
+width: 250px;
+height:35px;
+border: 1px #CFCFCF solid;
+margin:   ${props => props.theme.spacing(5)}px 0 ${props => props.theme.spacing(1.5)}px ${props => props.theme.spacing(1.5)}px ;
+padding-left: 0 ${props => props.theme.spacing(1.5)}px 
+`;
+
+const SearchButton = styled(Button)`
+width: 100px;
+margin:  ${props => props.theme.spacing(1.5)}px;
+display:block;
+color:white;
+`;
+
+const HiringImage = styled('object')`
+width: 250px;
+float:right;
+margin: 0 ${props => props.theme.spacing(10)}px ${props => props.theme.spacing(10)}px 0;
+`;
+
+const SearchSpinner = styled('div')`
+margin: ${props => props.theme.spacing(5)}px auto;
+width:50%;
+`;
+
+const FetchJobContent: React.FC = () => {
+  const [jobPosts, setJobPosts] = React.useState([]);
+  const [role, setRole] = React.useState('');
+  const [city, setCity] = React.useState('');
+  const [fetching, setFetching] = React.useState(false);
+  const handleChangeRole = React.useCallback((e) => setRole(e.target.value), []);
+  const handleChangeCity = React.useCallback((e) => setCity(e.target.value), []);
```

Comment:
I never use useCallback hook and I don't understand how it works. Can you explain how it works and why you use it here? It doesn't mean I suggest it is something wrong with this code. I just want to understand this solution :)

Reply:
I'm not 100% sure it is correct, but I use it to callback function (This hook is used to memoize functions, only for performance). When functions are passed to other components this hooks ,,limits"  us initializing the function every time the parent component is re-rendered.

Reply:
It is often hard to tell when to use performance optimization. Many experienced programmers recommend, as a rule of thumb, not doing any optimization until the app begins to slow down. 
But there are some cases when we will want to prevent future slowdown before it starts to show. These cases will usually concern preventing unneeded http requests or heavy computation (the latter not very likely in our case). 
In the case of `handleChangeRole/City` we don't have an http request or heavy computation, so I think we don't really need `useCallback`.
This article explains pretty well (in my opinion :) ) when we can try to optimize and when we shouldn't: https://dmitripavlutin.com/dont-overuse-react-usecallback/